api/trace/testtrace/span.go

```diff
@@ -69,6 +77,36 @@ func (s *Span) End(opts ...trace.EndOption) {
 	s.ended = true
 }
 
+func (s *Span) RecordError(ctx context.Context, err error, opts ...trace.ErrorOption) {
+	if err == nil || s.ended {
+		return
+	}
+
+	cfg := trace.ErrorConfig{}
+	for _, o := range opts {
+		o(&cfg)
+	}
+
+	if cfg.Timestamp.IsZero() {
+		cfg.Timestamp = time.Now()
+	}
+
+	if cfg.Status != codes.OK {
+		s.SetStatus(cfg.Status)
```

Comment:
This line raises questions for me about the `SetStatus` API, which is tangential to this PR.  The [Span.Status](https://github.com/open-telemetry/opentelemetry-proto/blob/5e1d5bc66d5a1fd9e0974d05ca11b8d0027cf5ea/opentelemetry/proto/trace/v1/trace.proto#L221) field has two fields, the code (passed here) and a message.  This should probably set the same `err.Error()` as used for the event message, but the `SetStatus` API doesn't accept a message string (which is a separate matter).

This has me wondering whether we want to use `RecordError(..., WithStatusCode())` as the conventional pattern for recording an error and setting the status. I'm not sure I'd pick this approach -- I'm inclined for there to be two methods, one to record an error event, and one to record an error event plus set the status (i.e., two methods `RecordErrorEvent` and `RecordErrorStatus`).